src/Illuminate/Collections/Collection.php

```diff
@@ -1729,8 +1729,10 @@ public function takeWhile($value)
     /**
      * Transform each item in the collection using a callback.
      *
-     * @param  callable(TValue, TKey): TValue  $callback
-     * @return $this
+     * @template TMapValue
+     *
+     * @param  callable(TValue, TKey): TMapValue  $callback
+     * @return $this<TKey, TMapValue>
```

Comment:
Both of these methods are incorrect:
- `$this<...>` is a syntax error
- need to use `@phpstan-this-out` for methods that mutate `$this`
- see: https://phpstan.org/r/5c9eb5c7-7570-4131-ac77-46d662ab74a6

Here is what they should be:


```suggestion
     * @param  callable(TValue, TKey): TMapValue  $callback
	 * @phpstan-this-out static<TKey, TMapValue>
     * @return $this
```